### PR TITLE
Add support for stage failures in pipelines

### PR DESCRIPTION
**Pull request by abhargav-mw**
This PR adds a new feature (xBug fix) for raising an exception if MATLAB process returned a non-zero exit code for Jenkins pipeline run. This adds a compatibility consideration as until now, plugin would set build result to failure incase MATLAB process returned non-zero exit code.

The behavior in case of MATLAB process being successful (that is returning 0 as exit code) stays the same.

### CODE REVIEW

**Review by mcafaro — APPROVED** [review]
:shipit:

File: src/main/resources/config.properties [hunk 1]
```diff
@@ -14,6 +14,7 @@ builder.matlab.customcommandoption.display.name = Custom
 Releaseinfo.matlab.version.not.found.error = Error finding MATLAB release for given MATLAB root. Verify MATLAB root path.
 matlab.not.found.error = Unable to launch MATLAB from the specified location. Verify the path to MATLAB root folder.
 matlab.not.found.error.for.node = Unable to launch MATLAB '%s' on the node '%s'. Verify global tool configuration for the specified node.
+matlab.execution.exception.prefix = Received non-zero exit code while trying to run MATLAB command (Exit code: %d)
```
Comment (mw-hrastega) [inline src/main/resources/config.properties:17]: Received a nonzero exit code %d while trying to run MATLAB.

We use "run MATLAB" in the doc. For example:
https://www.mathworks.com/help/matlab/matlab_env/run-matlab-from-azure-marketplace.html?searchHighlight=%22run%20matlab%22&s_tid=srchtitle

If you want MATLAB to be followed by something, you might consider "code" or "commands".

**Review by mw-hrastega — CHANGES_REQUESTED** [review]

**Review by nbhoski — APPROVED** [review]
Ship it !

**Review by mw-hrastega — APPROVED** [review]

**Review by jwpereira — APPROVED** [review]
🚢

**Pull request merged**